LongestSubstring.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package longestsubstring;
+
+/**
+ *
+ * @author kailabillie
+ */
+public class LongestSubstring {
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+        
+        System.out.println(lengthOfLongestSubstring("pwwkew"));
+    }
+    
+      public static int lengthOfLongestSubstring(String s) {
```

Comment:
I created a method using a similar design, so I hope you find this useful. It has a few differences from what you've already created.  
### Key Differences  
- Using StringBuilder, String, and int.
- removed char array and temp string
- included an else statement to handle repeating characters. 
- included a max int variable to keep track of longest current substring.
```
public static int lengthOfLongestSubstring(String input) {    
        StringBuilder sb = new StringBuilder();
        int max = -1;
        for (int i = 0; i<input.length(); i++) {
            if (sb.indexOf(input.substring(i,i+1)) == -1) 
                sb.append(input.charAt(i));
            else { 
                sb.deleteCharAt(sb.indexOf(input.substring(i,i+1)));
                sb.append(input.charAt(i));
            }
            max = max > sb.length() ? max : sb.length();           
        }  
        return max;
    }
```

I would include a check for null cases to avoid undesired behavior. You can always discuss what edge cases you should handle with the interviewer during the Talk phase.
```
if (Objects.equals(input,null)) throw new IllegalArgumentException();
```